stock_picking_invoice_link/models/sale_order.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2013-15 Agile Business Group sagl (<http://www.agilebg.com>)
+# Copyright 2017 Jacques-Etienne Baudoux <je@bcim.be>
+# Copyright 2021 Tecnativa - João Marques
+# Copyright 2022 Antony Herrera - LooErp
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from odoo import models
+from odoo.tools import float_compare
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def get_stock_moves_link_invoice(self):
+        return self.mapped("move_ids").filtered(
+            lambda mv: (
+                mv.state == "done"
+                and not (any(inv.state != "cancel" for inv in mv.invoice_line_ids.mapped("move_id")))
+                and not mv.scrapped
+                and (mv.location_dest_id.usage == "customer" or (mv.location_id.usage == "customer" and mv.to_refund))
+            )
+        )
+
+    def _prepare_invoice_line(self, **optional_values):
+        vals = super()._prepare_invoice_line(**optional_values)
+        stock_moves = self.get_stock_moves_link_invoice()
+        # Invoice returned moves marked as to_refund
+        if float_compare(self.qty_to_invoice, 0.0, precision_rounding=self.currency_id.rounding) < 0:
+            stock_moves = stock_moves.filtered(lambda m: m.to_refund and not m.invoice_line_ids)
+        vals["move_line_ids"] = [(4, m.id) for m in stock_moves]
```

Comment:
This not working
I create a Sale Order by 10U
delivery 10 U and create invoice, picking and invoice are linked

Now Create a return and try create new invoice(is create as refund) but picking is not asociate with refund

![image](https://user-images.githubusercontent.com/7775116/156676614-8314b3c1-e247-41fe-9f2e-67596e9be3bb.png)

No invoice related
![image](https://user-images.githubusercontent.com/7775116/156676649-9293e579-9f06-46a2-9900-7e899f04e693.png)